Services/RefundService.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+namespace MollieShopware\Services;
+
+use Doctrine\ORM\EntityManager;
+use Exception;
+use Mollie\Api\Resources\Order as MollieOrder;
+use Mollie\Api\Resources\Payment as MolliePayment;
+use Mollie\Api\Resources\OrderLine;
+use Mollie\Api\Resources\Refund;
+use MollieShopware\Models\OrderLines;
+use MollieShopware\Models\OrderLinesRepository;
+use MollieShopware\Models\Transaction;
+use MollieShopware\Traits\MollieApiClientTrait;
+use Shopware\Models\Order\Detail;
+use Shopware\Models\Order\Order;
+use Shopware\Models\Order\Status;
+
+class RefundService implements RefundInterface
+{
+    use MollieApiClientTrait;
+
+    /**
+     * @var EntityManager
+     */
+    private $modelManager;
+
+    public function __construct(EntityManager $modelManager)
+    {
+        $this->modelManager = $modelManager;
+    }
+
+    public function refundOrderAmount(Order $order, Transaction $transaction, float $customAmount = null)
+    {
+        $mollieClient = $this->getMollieApi($order->getShop()->getId());
+
+        if ($mollieClient === null) {
+            throw new \Exception('Something went wrong trying to get an API Client Instance. Please try again later.', 1);
+        }
+
+        if ($transaction->getMolliePaymentId() !== null) {
```

Comment:
i think $transaction should have something like "isOrderType" or such a getter
could you use that for these ifs?

Reply:
I could add that. But since we only need the IDs I think this would be a small overkill, isn't it?

Reply:
its just a getter in your existing object that does those checks internally.
i try to move to an internal oop approach instead of doing does checks outside
https://github.com/mollie/Shopware/blob/master/Models/Transaction.php#L432